readthedocs/templates/search/elastic_search.html

```diff
@@ -25,218 +27,219 @@
+          {% endblocktrans %}
+        </h3>
+      </div>
+
+      <div class="module-list">
+        <div class="module-list-wrapper">
+          <ul>
+
+            {% for result in results %}
+
+              <li class="module-item search-result-item">
+                <p class="module-item-title">
+                  {% if 'project' in result.meta.index %}
+                    <a href="{{ result.url }}">
+                      {{ result.name }} (<em>{{ result.slug }}</em>)
+                    </a>
+                    {% for fragment in result.meta.highlight.description %}
+                      <p class="fragment">
+                        ...{{ fragment|safe }}...
+                      </p>
+                    {% empty %}
+                      <p><em>No description for project.</em></p>
+                    {% endfor %}
+
+                  {% elif 'page' in result.meta.index %}
+                    <a href="{% doc_url result.project|get_project result.version result.full_path %}?highlight={{ query }}">
+                      {{ result.project }} - {% if result.meta.highlight.title %} {{ result.meta.highlight.title.0|safe }} {% else %} {{ result.title }} {% endif %}
+                    </a>
+
+                    {% for inner_hit in result.meta.inner_hits %}
+                      {% if inner_hit.type == 'domains' %}
+                        <p>
+                          <a href="{% doc_url result.project|get_project result.version inner_hit.source.doc_name %}#{{ inner_hit.source.anchor }}">
+                            {% if inner_hit.source.display_name|length >= 2 %}
+                              ({{ inner_hit.source.role_name }}) {{ inner_hit.source.display_name}}
+                            {% else %}
+                              {{ inner_hit.source.role_name }}
+                            {% endif %}
                           </a>
-                          {% for fragment in result.meta.highlight.description %}
-                            <p class="fragment">
-                            ...{{ fragment|safe }}...
-                            </p>
-                          {% empty %}
-                          <p><em>No description for project.</em></p>
-                          {% endfor %}
-
-                      {% elif 'domain' in result.meta.index %}
-
-                        <a href="{% doc_url result.project|get_project result.version result.doc_name %}?highlight={{ query }}#{{ result.anchor }}">
-                          {{ result.project }} - {% if result.meta.highlight.name|length %} {{ result.meta.highlight.name.0|safe }} {% else %} {{ result.name }} {% endif %}
-                        </a>
+                        </p>
                         <p class="fragment">
-                          {% if result.type_display %}
-                            {{ result.type_display|capfirst }}
+                          {% if inner_hit.highlight|get_key_or_none:"domains.name" %}
```

Comment:
Do we need `get_key_or_none` here? Doesn't Django handle the case where it's None?

Reply:
Yes.
Django throws an error otherwise.